File: backend/src/models/chapter.ts
import { Document, Schema, model } from "mongoose";

export interface IChapter extends Document {
    name: String;
    order: Number;
    description: String;
    summary: String;
    courseId: Schema.Types.ObjectId;
    noOfSubChapters: Number
}

const ChapterShema: Schema<IChapter> = new Schema({
    courseId: {
        type: Schema.Types.ObjectId,
        ref: 'Course',
        required: [true, "Course Id is required"]
    },
    name: {
        type: String,
        required: [true, "Chapter name is required"]
    },
    order:{
        type: Number,
        default: 0
    },
    description: {
        type: String,
        default: ""
    },
    summary: {
        type: String,
        required: [true, "Chapter summary is required"],
        default: ""
    },
    noOfSubChapters:{
        type: Number,
        default: 0
    }

},
    {
        timestamps: {
            createdAt: 'createdAt',
            updatedAt: 'updatedAt'
        }
    }
);

const Chapter = model<IChapter>("Chapter", ChapterShema);

export default Chapter;

File: new_mobile/lib/features/chapter/domain/entities/chapter.dart
import 'package:equatable/equatable.dart';

import '../../../features.dart';

class Chapter extends Equatable {
  final String id;
  final String courseId;
  final String name;
  final String description;
  final String summary;
  final int noOfSubchapters;
  final int order;
  final List<SubChapter>? subChapters;

  const Chapter({
    required this.noOfSubchapters,
    required this.id,
    required this.courseId,
    required this.name,
    required this.description,
    required this.summary,
    required this.order,
    this.subChapters,
  });

  @override
  List<Object?> get props => [
        id,
        courseId,
        name,
        description,
        summary,
        order,
        subChapters,
      ];
}
